sql/core/src/test/resources/sql-tests/inputs/array.sql

```diff
@@ -119,3 +119,15 @@ select get(array(1, 2, 3), 0);
 select get(array(1, 2, 3), 3);
 select get(array(1, 2, 3), null);
 select get(array(1, 2, 3), -1);
+
+-- function array_compact
+create temporary view invalid_datatype as select * from values
+(1), (2), (3)
+as invalid_datatype(id);
+
+select array_compact(id) from invalid_datatype;
+select array_compact(array("1", null, "2", null));
+select array_compact(array("1", null, "2", null));
+select array_compact(array(1D, null, 2D, null));
+select array_compact(array(array(1, 2, 3, null), null, array(4, null, 6)));
+select array_compact(array(null));
```

Comment:
Could you add a case that the input array without null ?

Reply:
Done